src/windows/volume.cc

```diff
@@ -65,6 +65,41 @@ HRESULT drivelist::volume::GetReadOnlyFlag(const wchar_t letter, BOOL *out) {
   return result;
 }
 
+HRESULT drivelist::volume::HasFileSystem(const wchar_t letter, BOOL *out) {
+  HANDLE handle = drivelist::volume::OpenHandle(letter, 0);
+  if (handle == INVALID_HANDLE_VALUE)
+    return E_HANDLE;
+
+  PARTITION_INFORMATION_EX information;
+  DWORD bytesReturned;
+
+  if (!DeviceIoControl(handle, IOCTL_DISK_GET_PARTITION_INFO_EX, NULL, 0,
+                       &information, sizeof(information),
+                       &bytesReturned, NULL)) {
+    CloseHandle(handle);
+    return E_FAIL;
+  }
+
+  // See https://msdn.microsoft.com/en-us/library/windows/desktop/aa365448(v=vs.85).aspx
+  switch (information.PartitionStyle) {
+  case 0:  // MBR
+    *out = IsRecognizedPartition(information.Mbr.PartitionType);
+    break;
+  case 1:  // GPT
+
+    // TODO(jviotti): The Windows API documents a constant called
+    // PARTITION_ENTRY_UNUSED_GUID, but I can't find its definition anywhere.
```

Comment:
Oh, I see, I misinterpreted your comment ;-)
I just did a bit of searching, and found it it `diskguid.h`
```
DEFINE_GUID(PARTITION_ENTRY_UNUSED_GUID,    0x00000000L, 0x0000, 0x0000, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00);    // Entry unused
```